File: swift/PraUTS/PraUTS/Mac.swift
//
//  Mac.swift
//  PraUTS
//
//  Created by Dodi Aditya on 06/11/23.
//

import Foundation

class Mac: OperatingSystem {
    private(set) var cycleCount: Int
    
    init(version: String, releaseDate: Date, language: String, serialNumber: String) {
        self.cycleCount = 0
        super.init(name: "MacOS", version: version, releaseDate: releaseDate, developer: "Apple", language: language, serialNumber: serialNumber)
    }
    
    convenience init(version: String, releaseDate: Date, language: String) {
        self.init(version: version, releaseDate: releaseDate, language: language, serialNumber: "")
    }
    
    override func printInfo() {
        super.printInfo()
        print("Cycle Count     : \(cycleCount)" )
    }
    
    func callSiri() {
        print("Hey Siri!")
    }
    
    func openAppStore() {
        print("App Store opened")
    }
    
    func spotlightSearch() {
        print("Search anything...")
    }
    
    func incrementCycleCount() {
        cycleCount += 1
    }
}

File: java/PraUTS/src/main/java/PraUTS/Mac.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package PraUTS;

import java.util.Date;

/**
 *
 * @author dodiaditya
 */
public class Mac extends OperatingSystem {
    private int cycle_count;
    
    public Mac(String version, Date release_date, String language) {
        this(version, release_date, language, null);
    }
    
    public Mac(String version, Date release_date, String language, String serial_number) {
        super("MacOS", version, release_date, "Apple", language, serial_number);
        this.cycle_count = 0;
    }
    
    @Override
    public void printAdditionalInfo() {
        System.out.println("Cycle Count     : " +  cycle_count);
    }
    
    public void incrementCycleCount() {
        this.cycle_count += 1;
    }
    
    public int getCycleCount() {
        return this.cycle_count;
    }
    
    public void callSiri() {
        System.out.println("Hey Siri!");
    }
    
    public void openAppStore() {
        System.out.println("App Store opened.");
    }
    
    public void spotlightSearch() {
        System.out.println("Search anything");
    }
}
